---
title: "Workflow Report"
subtitle: "hadrieng/arbetsprov"
output:
    revealjs::revealjs_presentation:
        highlight: pygments
        center: true
        transition: slide
        css: custom.css
        reveal_options:
            slideNumber: true
---

```{r load_pkgs, include=FALSE}
library(fastqcr)
library(tidyverse)
```

# Quality Control

## tools

- `fastqc` (quality control and plotting)
- `fastp` (quality filtering and trimming, adapter trimming)
- `fastqcr` (metrics aggregation and plotting)

```{r read_qc, include=FALSE}
qc <- qc_aggregate("./")
raw_file <- qc_read("IonTorrent_fastqc.zip")
trimmed_file <- qc_read("IonTorrentTrimmed_fastqc.zip")
```

## summary statistics


```{r echo=FALSE}
qc_stats(qc) %>%
    select(sample,
           GC = pct.gc,
           `n seq` = tot.seq,
           `length` = seq.length) %>% 
    knitr::kable()
```

## sequence quality


<div class="container">
<div class="col">
raw sequences
```{r echo=FALSE}
qc_plot(raw_file, "Per sequence quality scores")
```
</div>
<div class="col">
trimmed sequences
```{r echo=FALSE}
qc_plot(trimmed_file, "Per sequence quality scores")
```
</div>
</div>


## base quality


<div class="container">
<div class="col">
raw sequences
```{r echo=FALSE, warning=FALSE}
qc_plot(raw_file, "Per base sequence quality")
```
</div>
<div class="col">
trimmed sequences
```{r echo=FALSE, warning=FALSE}
qc_plot(trimmed_file, "Per base sequence quality")
```
</div>
</div>

## gc content


```{r echo=FALSE}
qc_plot(trimmed_file, "Per sequence GC content")
```


# Short Reads Assembly

```{r include=FALSE}
quast_report <- read_tsv("quast_results/latest/transposed_report.tsv")
```

## tools

- `spades` (assembly)
- `quast` (assembly qc)

##

```{r echo = FALSE}
quast_report %>%
    filter(Assembly == "IonTorrent") %>% 
    select(Assembly, `# contigs (>= 1000 bp)`, `Total length (>= 1000 bp)`,
           N50) %>% 
    knitr::kable()
```

# Taxonomy

## tools

- `mmseqs2` (taxonomy assignment. db: swissprot)
- `mlst_check` (MLST typing. db: pasteur)

```{r include=FALSE}
best_hit <- read_delim("best_hit.txt", delim = "\t", col_names = FALSE)$X1 %>% 
    as.character()
mlst <- read_tsv("mlst_results.allele.csv", skip = 5)
```

## Our mystery organism is

*`r best_hit`*

## mlst

```{r echo=FALSE}
mlst %>% 
    select(-`New ST`, -Contamination) %>% 
    t() %>%
    knitr::kable()
```

# Phylogeny

## tools

- `prodigal` (protein prediction)
- `cd-hit` (clustering)
- `mafft` (multiple sequence alignment)
- `fasttree` (tree building)

**Results not available :-(**

# Hybrid Assembly

## tools

- `spades` (assembly)
- `quast` (assembly qc)

##

```{r echo=FALSE}
quast_report %>%
    select(Assembly, `# contigs (>= 1000 bp)`, `Total length (>= 1000 bp)`,
           N50) %>% 
    knitr::kable()
```

##

```{r out.height = "460px", out.width='800px', echo=FALSE}
knitr::include_graphics("./quast_results/latest/basic_stats/cumulative_plot.pdf")
```

## {data-background-iframe="./quast_results/latest/icarus_viewers/contig_size_viewer.html"}

# Plasmids and Antibiotic resistance

## tools

- `abricate` (antibiotic resistance)
- `platon` (plasmid discovery)

```{r include=FALSE}
resistance <- read_tsv("hybrid.resistance.txt")
plasmids <- read_tsv("hybrid.contigs.tsv")
```

## Antiobiotic resistance

```{r echo=FALSE}
resistance %>% 
    select(-COVERAGE_MAP, -GAPS, -`#FILE`, -COVERAGE, -DATABASE,
           -ACCESSION, -`%COVERAGE`, -`%IDENTITY`, -PRODUCT)
```

## Plasmids

```{r echo=FALSE}
plasmids %>% 
    select(-RDS, -`Inc Type(s)`, -`# Replication`, -`# Mobilization`,
           -`# Conjugation`, -`# OriT`, -`# rRNAs`, -`# Plasmid Hits`)
```




